### PR TITLE
Update DB schema to version 2 (includes column for iKey)

### PR DESCRIPTION
**Pull request by ElektrojungeMS**

### CODE REVIEW

**Comment by jaelim-ms** [comment]
We need to add some unit tests to cover `iKey != nil` case. Giving reviewers visibility earlier to address any issues.

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m [hunk 1]
```diff
@@ -293,10 +311,16 @@ - (NSUInteger)countLogs {
 - (void)migrateDatabase:(void *)db fromVersion:(NSUInteger)version {
   if (version < 1) {
     NSString *migrationQuery = [NSString stringWithFormat:@"ALTER TABLE \"%@\" ADD COLUMN \"%@\" "
-                                                           "TEXT",
+                                                          "TEXT",
                                                           kMSLogTableName, kMSTargetTokenColumnName];
     [MSDBStorage executeNonSelectionQuery:migrationQuery inOpenedDatabase:db];
   }
+  if (version < 2) {
```
Comment (dhei) [inline AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: Please don't use magic number

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m [hunk 2]
```diff
@@ -293,10 +311,16 @@ - (NSUInteger)countLogs {
 - (void)migrateDatabase:(void *)db fromVersion:(NSUInteger)version {
   if (version < 1) {
```
Comment (dhei) [inline AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: Maybe making version `1` as a constant?

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaelim-ms) [reply AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: You don't want to duplicate migration code multiple times and it only takes one query at a time as you did in Android implementation.

File: AppCenter/AppCenter/Internals/Storage/MSStorage.h [hunk 3]
```diff
@@ -49,15 +49,18 @@ typedef void (^MSLoadDataCompletionBlock)(NSArray<id<MSLog>> *_Nullable logArray
 
 /**
  * Return the most recent logs for a Group Id.
- *
+ *
```
Comment (achocron) [inline AppCenter/AppCenter/Internals/Storage/MSStorage.h]: looks like an extra space was added

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m [hunk 4]
```diff
@@ -80,25 +89,35 @@ - (BOOL)saveLog:(id<MSLog>)log withGroupId:(NSString *)groupId {
 
 - (BOOL)loadLogsWithGroupId:(NSString *)groupId
                       limit:(NSUInteger)limit
-             withCompletion:(nullable MSLoadDataCompletionBlock)completion {
+                      iKeys:(nullable NSArray< NSString *> *)iKeys
+          completionHandler:(nullable MSLoadDataCompletionBlock)completionHandler {
```
Comment (clpolet) [inline AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: It probably makes sense to rename `MSLoadDataCompletionBlock` to `MSLoadDataCompletionHandler` now that `completion` is renamed `completionHandler`.

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m [hunk 5]
```diff
@@ -52,13 +53,21 @@ - (BOOL)saveLog:(id<MSLog>)log withGroupId:(NSString *)groupId {
                                  kMSGroupIdColumnName, kMSLogColumnName, groupId, base64Data];
 
   // Serialize target token.
-  if ([(NSObject *)log isKindOfClass:[MSCommonSchemaLog class]]) {
+  if ([(NSObject *) log isKindOfClass:[MSCommonSchemaLog class]]) {
     NSString *targetToken = [[log transmissionTargetTokens] anyObject];
     NSString *encryptedToken = [self.targetTokenEncrypter encryptString:targetToken];
+    NSString *iKey = ((MSCommonSchemaLog *) log).iKey;
```
[the remark this answers is not in the document]
Comment (jaelim-ms) [reply AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: Above two lines are also dealing with MSCommonSchemaLog specific data. Just in the order of SQL statement.
